nixos/modules/services/web-apps/rutorrent.nix

```diff
@@ -0,0 +1,281 @@
+                $overwriteUploadedTorrents = false;     // Overwrite existing uploaded torrents in profile/torrents directory or make unique name
+
+                $topDirectory = '/';			// Upper available directory. Absolute path with trail slash.
+                $forbidUserSettings = false;
+
+                $scgi_port = 0;
+                $scgi_host = "unix://${cfg.rpcSocket}";
+
+                $XMLRPCMountPoint = "/RPC2";		// DO NOT DELETE THIS LINE!!! DO NOT COMMENT THIS LINE!!!
+
+                $pathToExternals = array(
+                  "php" 	=> ''',			// Something like /usr/bin/php. If empty, will be found in PATH.
+                  "curl"	=> ''',			// Something like /usr/bin/curl. If empty, will be found in PATH.
+                  "gzip"	=> ''',			// Something like /usr/bin/gzip. If empty, will be found in PATH.
+                  "id"	=> ''',			// Something like /usr/bin/id. If empty, will be found in PATH.
+                  "stat"	=> ''',			// Something like /usr/bin/stat. If empty, will be found in PATH.
+                );
+
+                $localhosts = array( 			// list of local interfaces
+                  "127.0.0.1",
+                  "localhost",
+                );
+
+                $profilePath = '../share';		// Path to user profiles
+                $profileMask = 0777;			// Mask for files and directory creation in user profiles.
+                // Both Webserver and rtorrent users must have read-write access to it.
+                // For example, if Webserver and rtorrent users are in the same group then the value may be 0770.
+
+                $tempDirectory = null;			// Temp directory. Absolute path with trail slash. If null, then autodetect will be used.
+
+                $canUseXSendFile = false;		// If true then use X-Sendfile feature if it exist
+
+                $locale = "UTF8";
+            '';
+          in {
+            wantedBy = [ "multi-user.target" ];
+            before = [ "phpfpm-rutorrent.service" ];
+            script = ''
+              ln -sf ${pkgs.rutorrent}/{css,images,js,lang,index.html} ${cfg.home}/
+              mkdir -p ${cfg.home}/{conf,logs,plugins} ${cfg.home}/share/{settings,torrents,users}
+              ln -sf ${pkgs.rutorrent}/conf/{access.ini,plugins.ini} ${cfg.home}/conf/
+              ln -sf ${rutorrentConfig} ${cfg.home}/conf/config.php
+
+              cp -r ${pkgs.rutorrent}/php ${cfg.home}/
+
+              ${optionalString (cfg.plugins != [])
+                ''cp -r ${concatMapStringsSep " " (p: "${pkgs.rutorrent}/plugins/${p}") cfg.plugins} ${cfg.home}/plugins/''}
+
+              chown -R rutorrent:rutorrent ${cfg.home}/{conf,share,logs,plugins}
+              chmod -R 755 ${cfg.home}/{conf,share,logs,plugins}
+            '';
+            serviceConfig.Type = "oneshot";
+          };
+        };
+
+        tmpfiles.rules = [ "d '${cfg.home}' 0775 rutorrent rutorrent -" ];
```

Comment:
I don't think we want this to be tmp beacuse it also hold persistent state like user settings. It would be better to move temporary data to `/run` with `RuntimeDirectory=rutorrent`, `RuntimeDirectoryPreserve=no` and `PartOf=phpfpm-rutorrent.service`.